confmap/resolver.go

```diff
@@ -39,6 +39,9 @@ type ResolverSettings struct {
 	// It is required to have at least one factory.
 	ProviderFactories []ProviderFactory
 
+	// ProviderModules maps provider types to their respective go modules.
+	ProviderModules map[string]string
```

Comment:
The PR in general LGTM but I don't feel like this is the best place for this field. The resolver does not really use this field. Should this be in `otelcol.CollectorSettings` maybe?